src/server/pkg/sync/sync.go

```diff
@@ -191,3 +203,151 @@ func PushObj(pachClient pachclient.APIClient, commit *pfs.Commit, objClient obj.
 	}
 	return eg.Wait()
 }
+
+// PullObj pulls data from an object store into a pfs commit.
+func PullObj(pachClient pachclient.APIClient, commit *pfs.Commit, objClient obj.Client, root string) error {
+	var eg errgroup.Group
+	sem := make(chan struct{}, semCap)
+	if err := objClient.Walk(root, func(name string) error {
+		sem <- struct{}{}
```

Comment:
Would you be okay adding a comment here that 'sem' throttles the reader in 'walk'? It just took me a minute to figure out what was going on here.